driver-core/src/main/com/mongodb/connection/ClusterSettings.java

```diff
@@ -629,9 +630,14 @@ private ClusterSettings(final Builder builder) {
             if (builder.mode == ClusterConnectionMode.SINGLE && builder.hosts.size() > 1) {
                 throw new IllegalArgumentException("Can not directly connect to more than one server");
             }
-            mode = builder.mode != null ? builder.mode : hosts.size() == 1 ? ClusterConnectionMode.SINGLE : ClusterConnectionMode.MULTIPLE;
+            if (builder.mode != null) {
```

Comment:
If this conditional is removed, the only test that fails is `when connection string is applied to builder, all properties should be set`.  It seems like there should be a more direct test which covers the scenario where there is no connection string applied at all.